From 34024c9b696b9d650bd4e9d517b6cbf07d6117c3 Mon Sep 17 00:00:00 2001
From: Chris Wilson <chris@chris-wilson.co.uk>
Date: Fri, 10 Sep 2010 22:33:19 +0100
Patch-mainline: 2.6.37
References: fate#310916
Git-commit: d5e0d2f51977fe1f7fd6ee5c1a4476b43bad8f92
Subject: [PATCH 1384/2588] drm/i915: Ensure all PLL registers are flushed
 before a udelay()

Signed-off-by: Chris Wilson <chris@chris-wilson.co.uk>
(cherry picked from commit d5e0d2f51977fe1f7fd6ee5c1a4476b43bad8f92)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/intel_display.c |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index 8b5b935..0999d4b 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -1655,6 +1655,7 @@ static void ironlake_set_pll_edp (struct drm_crtc *crtc, int clock)
 		dpa_ctl |= DP_PLL_FREQ_270MHZ;
 	}
 	I915_WRITE(DP_A, dpa_ctl);
+	POSTING_READ(DP_A);
 
 	udelay(500);
 }
@@ -1722,6 +1723,7 @@ static void ironlake_fdi_link_train(struct drm_crtc *crtc)
 	temp &= ~FDI_LINK_TRAIN_NONE;
 	temp |= FDI_LINK_TRAIN_PATTERN_2;
 	I915_WRITE(fdi_rx_reg, temp);
+	POSTING_READ(fdi_rx_reg);
 	udelay(150);
 
 	tries = 0;
@@ -1802,6 +1804,7 @@ static void gen6_fdi_link_train(struct drm_crtc *crtc)
 		temp &= ~FDI_LINK_TRAIN_VOL_EMP_MASK;
 		temp |= snb_b_fdi_train_param[i];
 		I915_WRITE(fdi_tx_reg, temp);
+		POSTING_READ(fdi_tx_reg);
 		udelay(500);
 
 		temp = I915_READ(fdi_rx_iir_reg);
@@ -1837,6 +1840,7 @@ static void gen6_fdi_link_train(struct drm_crtc *crtc)
 		temp |= FDI_LINK_TRAIN_PATTERN_2;
 	}
 	I915_WRITE(fdi_rx_reg, temp);
+	POSTING_READ(fdi_rx_reg);
 	udelay(150);
 
 	for (i = 0; i < 4; i++ ) {
@@ -1844,6 +1848,7 @@ static void gen6_fdi_link_train(struct drm_crtc *crtc)
 		temp &= ~FDI_LINK_TRAIN_VOL_EMP_MASK;
 		temp |= snb_b_fdi_train_param[i];
 		I915_WRITE(fdi_tx_reg, temp);
+		POSTING_READ(fdi_tx_reg);
 		udelay(500);
 
 		temp = I915_READ(fdi_rx_iir_reg);
-- 
1.7.6

